cfgov/unprocessed/js/molecules/GlobalSearch.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+// Required polyfills for <IE9.
+require( '../modules/polyfill/query-selector' );
+require( '../modules/polyfill/event-listener' );
+require( '../modules/polyfill/class-list' );
+
+// Required modules.
+var atomicCheckers = require( '../modules/util/atomic-checkers' );
+var breakpointState = require( '../modules/util/breakpoint-state' );
+
+/**
+ * GlobalSearch
+ * @class
+ *
+ * @classdesc Initializes a new GlobalSearch molecule.
+ *
+ * @param {HTMLNode} element
+ *   The DOM element within which to search for the molecule.
+ * @returns {Object} An GlobalSearch instance.
+ */
+function GlobalSearch( element ) {
+
+  var BASE_CLASS = 'm-global-search';
+
+  var _dom =
+    atomicCheckers.validateDomElement( element, BASE_CLASS, 'GlobalSearch' );
+  var _triggerDom = _dom.querySelector( '.' + BASE_CLASS + '_trigger' );
+  var _contentDom = _dom.querySelector( '.' + BASE_CLASS + '_content' );
+  var _searchInputDom;
+  var _clearBtnDom;
+
+  var _isExpanded = false;
+
+  /**
+   * @returns {Object} The GlobalSearch instance.
+   */
+  function init() {
+    var searchInputSelector = '.' + BASE_CLASS + '_content-form input';
+    var searchIconSelector =
+      '.' + BASE_CLASS + ' .input-contains-label_before__search';
+    var clearBtnSelector =
+      '.' + BASE_CLASS + ' .input-contains-label_after__clear';
+
+    _searchInputDom = _contentDom.querySelector( searchInputSelector );
+    _clearBtnDom = _contentDom.querySelector( clearBtnSelector );
+    var searchIconDom = _contentDom.querySelector( searchIconSelector );
+
+    _triggerDom.addEventListener( 'click', _triggerClicked );
+    _clearBtnDom.addEventListener( 'click', _clearClicked );
+    _searchInputDom.addEventListener( 'keyup', _inputTyped );
+    searchIconDom.addEventListener( 'click', _searchLabelClicked );
+
+    _setClearBtnState( _searchInputDom.value );
+
+    return this;
+  }
+
+  /**
+   * Event handler for when the search icon is clicked in the
+   * expanded state at desktop sizes. Closes the search box.
+   */
+  function _searchLabelClicked() {
```

Comment:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/4ff9279128485d9d686ed1b9d4456295cced6c62